code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingPropositionItem.java

```diff
@@ -113,82 +100,137 @@ void track(final PropositionEventType interaction) {
     }
 
     /**
-     * Creates an {@code Inbound} object from this {@code PropositionItem}'s content.
+     * Creates a {@code MessagingPropositionItem} object from the provided {@code RuleConsequence}.
      *
-     * @return {@link Inbound} object created from this {@link MessagingPropositionItem}'s content.
+     * @return {@link MessagingPropositionItem} object created from the provided {@link RuleConsequence}.
      */
-    public Inbound decodeContent() {
-        Inbound inboundContent = null;
+    public static MessagingPropositionItem fromRuleConsequence(final RuleConsequence consequence) {
```

Comment:
This method returns the same type of object as the class it is in. Can we refactor it to be constructor instead?
`public MessagingPropositionItem(final RuleConsequence consequence) {
              // get uniqueId, schema and itemData from consequence
              this(uniqueId, schema, itemData);
}`

Reply:
i think I prefer to keep the static method as there may be some confusion if i use two different constructors for the object.